valhalla/filesystem.h

```diff
@@ -37,12 +37,14 @@ class path {
     // TODO: squash repeated separators
     // delineate the path
     for (size_t npos = path_name_.find_first_of(preferred_separator); npos != std::string::npos;
-         npos = path_name_.find_first_of(preferred_separator, npos + 1))
+         npos = path_name_.find_first_of(preferred_separator, npos + 1)) {
```

Comment:
our formatter does not enforce this right? this was just a personal nitpick im guessing? (i hate the extra braces so i always leave them off :smile: )

Reply:
Actually, clang-tidy made this change

Reply:
`master` already has a `clang-tidy` config set to add these extra braces via `readability-braces-around-statements` in  https://github.com/valhalla/valhalla/blob/master/.clang-tidy#L3

Reply:
I think this means clang-tidy is misconfigured on master, since these missing braces should have failed the lint job, right?

Reply:
I believe so. It's hard to verify since the master branch uses `run-clang-tidy-5.0.py` which doesn't seem part of the mason install nor recent versions of clang-format.

Reply:
maybe it actually means the CI doesnt mark it as a failable offense. what sucks is that my IDE doesnt have an auto add braces thing, which means ill need to run this manually or actually learn to type all this extra cruft :stuck_out_tongue:

Reply:
honestly though, how on earth is this the only file in the project that has this. i leave off curly braces *EVERYWHERE* i feel like maybe the check isnt working in other places if it only caught this one file that is at the top of the include hierarchy. is it like not running recursively or something?

Reply:
So I think I introduced a change here just to get a diff to trigger. Remember, this new tidy script only runs on modified files in order to finish faster.

Reply:
If you run the script locally, it'll apply the fixes for you. You only need to commit the change.

Reply:
If I introduce a change to bidirectional, I get the following output:
```bash
scripts/clang-tidy-only-diff.sh
* Already installed at /home/anders/Documents/Programming/mapbox/valhalla/mason_packages/linux-x86_64/clang-tidy/7.0.0
* Linking /home/anders/Documents/Programming/mapbox/valhalla/mason_packages/linux-x86_64/clang-tidy/7.0.0
* Links will be inside /home/anders/Documents/Programming/mapbox/valhalla/mason_packages/.link/
* Using bash fallback for symlinking (install lndir for faster symlinking)
* Done linking /home/anders/Documents/Programming/mapbox/valhalla/mason_packages/linux-x86_64/clang-tidy/7.0.0
Detected changes in /home/anders/Documents/Programming/mapbox/valhalla/src/thor/bidirectional_astar.cc
/home/anders/Documents/Programming/mapbox/valhalla/src/thor/bidirectional_astar.cc:162:87: warning: statement should be inside braces [readability-braces-around-statements]
      } else if (!hierarchy_limits_forward_[trans->endnode().level()].StopExpanding())
                                                                                      ^
                                                                                       {
/home/anders/Documents/Programming/mapbox/valhalla/src/thor/bidirectional_astar.cc:162:87: note: FIX-IT applied suggested code changes
/home/anders/Documents/Programming/mapbox/valhalla/src/thor/bidirectional_astar.cc:164:100: note: FIX-IT applied suggested code changes
            ExpandForward(graphreader, trans->endnode(), pred, pred_idx, true) || found_valid_edge;
                                                                                                   ^
3174 warnings generated.
clang-tidy applied 2 of 2 suggested fixes.
Suppressed 3173 warnings (3173 in non-user code).
Use -header-filter=.* to display errors from all non-system headers. Use -system-headers to display errors from system headers as well.
Tidy introduced changes
```

Notably `Detected changes in /home/anders/Documents/Programming/mapbox/valhalla/src/thor/bidirectional_astar.cc`

Reply:
this is clearly different than before though. its fine to run it only on changed files but the effect here is that diffs will start to end up being formatting for every file that we change. diffs for peoples work are hard enough to read and make sense of without format changes that are unrelated to functionality changes. in a big project like this i think we need to strive to make prs that are as small and as pointed to the actual problem being sovled as possible.

i get that its easy to run it and then commit the changes. i personally think that its not worth it if its basically formatting the code. that is what clang-format is for. tidy should catch stuff like unitialized members, unhandled enumeration values in switch cases and the like.

is it just the case that for whatever reason tidy never returned non 0 exit code before? and if so, why? or is it that CI was not configured to care if it did?

ok i just tried it on master and yeah... it changes a lot...

```
	geändert:       ../src/baldr/compression_utils.cc
	geändert:       ../src/baldr/curler.cc
	geändert:       ../src/baldr/datetime.cc
	geändert:       ../src/baldr/graphreader.cc
	geändert:       ../src/baldr/graphtile.cc
	geändert:       ../src/baldr/tz_alt.cpp
	geändert:       ../src/bindings/node/node.cc
	geändert:       ../src/loki/locate_action.cc
	geändert:       ../src/loki/matrix_action.cc
	geändert:       ../src/loki/reach.cc
	geändert:       ../src/loki/route_action.cc
	geändert:       ../src/loki/search.cc
	geändert:       ../src/loki/trace_route_action.cc
	geändert:       ../src/loki/worker.cc
	geändert:       ../src/meili/map_matcher_factory.cc
	geändert:       ../src/meili/viterbi_search.cc
	geändert:       ../src/midgard/pointll.cc
	geändert:       ../src/midgard/polyline2.cc
	geändert:       ../src/midgard/tiles.cc
	geändert:       ../src/mjolnir/admin.cc
	geändert:       ../src/mjolnir/countryaccess.cc
	geändert:       ../src/mjolnir/directededgebuilder.cc
	geändert:       ../src/mjolnir/graphbuilder.cc
	geändert:       ../src/mjolnir/graphenhancer.cc
	geändert:       ../src/mjolnir/graphtilebuilder.cc
	geändert:       ../src/mjolnir/linkclassification.cc
	geändert:       ../src/mjolnir/osmdata.cc
	geändert:       ../src/mjolnir/pbfgraphparser.cc
	geändert:       ../src/mjolnir/servicedays.cc
	geändert:       ../src/mjolnir/valhalla_add_predicted_traffic.cc
	geändert:       ../src/odin/enhancedtrippath.cc
	geändert:       ../src/odin/maneuver.cc
	geändert:       ../src/sif/autocost.cc
	geändert:       ../src/sif/bicyclecost.cc
	geändert:       ../src/sif/dynamiccost.cc
	geändert:       ../src/sif/motorcyclecost.cc
	geändert:       ../src/sif/motorscootercost.cc
	geändert:       ../src/sif/pedestriancost.cc
	geändert:       ../src/sif/truckcost.cc
	geändert:       ../src/thor/map_matcher.cc
	geändert:       ../src/thor/route_action.cc
	geändert:       ../src/thor/route_matcher.cc
	geändert:       ../src/thor/trace_route_action.cc
	geändert:       ../src/thor/worker.cc
	geändert:       ../src/tile_server.cc
	geändert:       ../src/tyr/route_serializer_osrm.cc
	geändert:       ../src/valhalla_path_comparison.cc
	geändert:       ../src/valhalla_run_isochrone.cc
	geändert:       ../src/valhalla_run_matrix.cc
	geändert:       ../src/valhalla_run_route.cc
	geändert:       ../src/worker.cc
	geändert:       ../test/aabb2.cc
	geändert:       ../test/access_restriction.cc
	geändert:       ../test/admin.cc
	geändert:       ../test/attributes_controller.cc
	geändert:       ../test/complexrestriction.cc
	geändert:       ../test/compression.cc
	geändert:       ../test/countryaccess.cc
	geändert:       ../test/datetime.cc
	geändert:       ../test/directededge.cc
	geändert:       ../test/distanceapproximator.cc
	geändert:       ../test/double_bucket_queue.cc
	geändert:       ../test/edgecollapser.cc
	geändert:       ../test/edgeinfobuilder.cc
	geändert:       ../test/edgestatus.cc
	geändert:       ../test/encode.cc
	geändert:       ../test/filesystem.cc
	geändert:       ../test/graphid.cc
	geändert:       ../test/graphparser.cc
	geändert:       ../test/graphreader.cc
	geändert:       ../test/graphtile.cc
	geändert:       ../test/graphtilebuilder.cc
	geändert:       ../test/graphtileheader.cc
	geändert:       ../test/grid_traversal.cc
	geändert:       ../test/gridded_data.cc
	geändert:       ../test/http_tiles.cc
	geändert:       ../test/idtable.cc
	geändert:       ../test/instructions.cc
	geändert:       ../test/json.cc
	geändert:       ../test/linesegment2.cc
	geändert:       ../test/location.cc
	geändert:       ../test/logging.cc
	geändert:       ../test/maneuversbuilder.cc
	geändert:       ../test/mapmatch.cc
	geändert:       ../test/matrix.cc
	geändert:       ../test/minbb.cc
	geändert:       ../test/multipoint_routes.cc
	geändert:       ../test/names.cc
	geändert:       ../test/narrativebuilder.cc
	geändert:       ../test/node_search.cc
	geändert:       ../test/nodeinfo.cc
	geändert:       ../test/nodetransition.cc
	geändert:       ../test/openlr.cc
	geändert:       ../test/parse_request.cc
	geändert:       ../test/point2.cc
	geändert:       ../test/pointll.cc
	geändert:       ../test/polyline2.cc
	geändert:       ../test/predictedspeeds.cc
	geändert:       ../test/predictive_traffic.cc
	geändert:       ../test/reach.cc
	geändert:       ../test/recover_shortcut.cc
	geändert:       ../test/refs.cc
	geändert:       ../test/sample.cc
	geändert:       ../test/search.cc
	geändert:       ../test/sequence.cc
	geändert:       ../test/servicedays.cc
	geändert:       ../test/shape_attributes.cc
	geändert:       ../test/sign.cc
	geändert:       ../test/signinfo.cc
	geändert:       ../test/skadi_service.cc
	geändert:       ../test/streetname.cc
	geändert:       ../test/streetname_us.cc
	geändert:       ../test/streetnames.cc
	geändert:       ../test/streetnames_us.cc
	geändert:       ../test/thor_service.cc
	geändert:       ../test/thor_worker.cc
	geändert:       ../test/tilehierarchy.cc
	geändert:       ../test/tiles.cc
	geändert:       ../test/timedep_paths.cc
	geändert:       ../test/timeparsing.cc
	geändert:       ../test/transitdeparture.cc
	geändert:       ../test/transitroute.cc
	geändert:       ../test/transitschedule.cc
	geändert:       ../test/transitstop.cc
	geändert:       ../test/trivial_paths.cc
	geändert:       ../test/turnlanes.cc
	geändert:       ../test/uniquenames.cc
	geändert:       ../test/util_midgard.cc
	geändert:       ../test/util_odin.cc
	geändert:       ../test/util_skadi.cc
	geändert:       ../test/utrecht.cc
	geändert:       ../test/vector2.cc
	geändert:       ../test/verbal_text_formatter.cc
	geändert:       ../test/verbal_text_formatter_us.cc
	geändert:       ../test/verbal_text_formatter_us_co.cc
	geändert:       ../test/verbal_text_formatter_us_tx.cc
	geändert:       ../test/viterbi_search.cc
	geändert:       ../valhalla/meili/viterbi_search.h
```

if we want to do this we should do all the files at once so we dont have to keep looking at meaningless changes all the time. and if we havent already done so we should make sure CI fails when tidy makes changes to the code similar to what the formatter script does. i havent researched it but i expect its going to be harder to get IDEs to automatically tidy.

Reply:
oh man even worse i looked at the diff that clang tidy made... i see fun like this:

```diff
-  if (!begin_node)
-    return {shortcut_id};
+  if (!begin_node) {
+    return {shortcut_id
+    }
+  };
```

seems clang tidy doesnt understand curly brace initialization... after seeing this i think we should actually remove this check from clang tidy.

Reply:
> oh man even worse i looked at the diff that clang tidy made... i see fun like this:
> 
> ```diff
> -  if (!begin_node)
> -    return {shortcut_id};
> +  if (!begin_node) {
> +    return {shortcut_id
> +    }
> +  };
> ```
> 
> seems clang tidy doesnt understand curly brace initialization... after seeing this i think we should actually remove this check from clang tidy.

Yeah not a fan of the above.

Reply:
I'm happy to submit an additional commit that disables `readability-braces-around-statements` from the existing `clang-tidy` statement. It sounds we're leaning to not using the braces for one-line conditionals.

Reply:
ship!